use-timescale/migration/dual-write-and-backfill.md

```diff
@@ -0,0 +1,245 @@
+of TimescaleDB must be the same.
+
+Dual-write and backfill works well when:
+1. The bulk of the (on-disk) data is in time-series tables.
+1. Writes by the application do not reference historical time-series data.
+1. There is no requirement for transactional consistency (that is, it is possible
+   to filter the time-series data by time and retain data integrity).
+1. No `UPDATE` or `DELETE` queries will be run on time-series data in the
+   source database during the migration process (or if they are, it happens in
+   a controlled manner, such that it's possible to either ignore, or
+   re-backfill).
+1. Either the relational (non-time-series) data is small enough to be copied
+   from source to target in an acceptable amount of time for this to be done
+   with downtime, or the relational data can be copied asynchronously while the
+   application continues to run (that is, changes relatively infrequently).
+
+## Migration process
+
+In detail, the migration process consists of the following steps:
+1. Set up a second database
+1. Modify the application to write to a secondary database
+1. Migrate schema and relational data from source to target
+1. Start the application in dual-write mode
+1. Determine the consistency time `T`
+1. Backfill time-series data from source to target
+1. Enable retention and compression policies
+1. Validate that all data is present in target database
+1. Validate that target database can handle production load
+1. Switch application to treat target database as primary (potentially continuing to write into source database, as a backup)
+
+### 1. Set up a second database
+
+[Create a database service in Timescale][create-service].
+
+### 2. Modify the application to write to a secondary database 
+
+How exactly to do this is dependent on the language that your application is
+written in, and on how exactly your ingestion and application function. In the
+simplest case, you simply execute two inserts in parallel. In the general case,
+you will need to think about how to handle the failure to write to either the
+old or new database, and what mechanism you want to or can build to recover
+from such a failure.
+
+You may also want to execute the same read queries on the old and new database,
+in order to evaluate the correctness and performance of the results which the
+queries deliver. Bear in mind that the new database will spend a certain amount
+of time without all data being present, so you should expect that the results
+are not the same for some period (potentially a number of days).
+
+### 3. Set up schema and migrate relational data to new database
+
+How exactly you perform this is dependent on whether you're migrating from
+plain PostgreSQL, TimescaleDB, or some other database.
+
+#### From TimescaleDB
+
+Dump the database roles from the source database:
+
+```
+pg_dumpall -d "$SOURCE" \
+  --quote-all-identifiers \
+  --roles-only \
+  --file=roles.sql
+```
+
+Fix up the dumped roles:
+
+```
+sed -i -E \
```

Comment:
We need to add a note about this being gnu/sed , maybe we can make it compatible with both.

Reply:
with bsd sed (the one used in macos) it needs to be `sed -i '' -E`